src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -43,14 +43,18 @@ private BoundExpression RewriteCollectionExpressionConversion(Conversion convers
                 switch (collectionTypeKind)
                 {
                     case CollectionExpressionTypeKind.ImplementsIEnumerable:
-                        if (ConversionsBase.IsSpanOrListType(_compilation, node.Type, WellKnownType.System_Collections_Generic_List_T, out var listElementType))
+                        if (ConversionsBase.IsSpanOrListType(_compilation, node.Type, WellKnownType.System_Collections_Generic_List_T, out var listElementType) &&
+                            usesParameterlessListConstructor(_compilation, node))
```

Comment:
>usesParameterlessListConstructor(_compilation, node)) [](http://example.com/codeflow?start=28&length=53)

The local function `usesParameterlessListConstructor()` was extracted from `useListOptimization()` and used here, to avoid calling `TryRewriteSingleElementSpreadToList()` when arguments are included with a `List<T>` target type. See `List_SingleSpread()` unit test which would fail to set `capacity:` in `[with(capacity: capacity), ..e]` otherwise.